sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/operator/BeamSqlCaseExpression.java

```diff
@@ -51,7 +51,8 @@ public BeamSqlCaseExpression(List<BeamSqlExpression> operands) {
 
   @Override public BeamSqlPrimitive evaluate(BeamRecord inputRow, BoundedWindow window) {
     for (int i = 0; i < operands.size() - 1; i += 2) {
-      if (opValueEvaluated(i, inputRow, window)) {
+      Boolean b = opValueEvaluated(i, inputRow, window);
+      if (b != null && b) {
```

Comment:
@xumingming @akedin Do you think this change is correct?

Reply:
Looks correct. From what I see this will still be valid:
*`CASE a WHEN null THEN b ELSE c` => `c` for all `a`, including `a == null`. 
*`NULLIF(a,null)` => `null` for all `a` including `a == null`
*`COALESCE(a,b,c,d)` => first non-null among all operands